### ISSUE TITLE
Failed to start tidb: timed out waiting for port 4000 to be started after 2m0s

### ISSUE THREAD

**Issue by shenoy-anurag** [pingcap/tidb#20545]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

OS: Ubuntu 18.04
Docs followed: https://docs.pingcap.com/tidb/stable/production-deployment-using-tiup
Deployed basic topology to remote server.

Topology:
1 pd, 1 Tidb, 3 TiKV, 1 TiFlash all on single machine for testing.

### 2. What did you expect to see? (Required)

Starting component tidb
	Starting instance tidb 127.0.0.1:4000
	Start tidb 127.0.0.1:4000 success

### 3. What did you see instead (Required)

Error: failed to start tidb: 	tidb 154.27.78.227:4000 failed to start: timed out waiting for port 4000 to be started after 2m0s, please check the log of the instance: timed out waiting for port 4000 to be started after 2m0s

Verbose debug logs has been written to /home/user/Desktop/logs/...

Error on control machine:
`INFO Execute command finished {"code": 1, "error": "failed to start tidb: \ttidb 127.0.0.1:4000 failed to start: timed out waiting for port 4000 to be started after 2m0s, please check the log of the instance: timed out waiting for port 4000 to be started after 2m0s", "errorVerbose": "timed out waiting for port 4000 to be started after 2m0s\ngithub.com/pingcap/tiup/pkg/cluster/module.(*WaitFor).Execute\n\tgithub.com/pingcap/tiup@/pkg/cluster/module/wait_for.go:90\ngithub.com/pingcap/tiup/pkg/cluster/spec.PortStarted\n\tgithub.com/pingcap/tiup@/pkg/cluster/spec/instance.go:101\ngithub.com/pingcap/tiup/pkg/cluster/spec.(*BaseInstance).Ready\n\tgithub.com/pingcap/tiup@/pkg/cluster/spec/instance.go:132\ngithub.com/pingcap/tiup/pkg/cluster/operation.startInstance\n\tgithub.com/pingcap/tiup@/pkg/cluster/operation/action.go:563\ngithub.com/pingcap/tiup/pkg/cluster/operation.StartComponent.func1\n\tgithub.com/pingcap/tiup@/pkg/cluster/operation/action.go:678\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357\n\ttidb 127.0.0.1:4000 failed to start: timed out waiting for port 4000 to be started after 2m0s, please check the log of the instance\nfailed to start tidb"}`

TiDB logs on server:
`[error="other error: [components/tidb_query/src/expr/ctx.rs:87]: unknown or incorrect time zone: posixrules"] [2020/10/20 07:24:51.093 -04:00] [FATAL] [terror.go:257] ["unexpected error"] [error="other error: [components/tidb_query/src/expr/ctx.rs:87]: unknown or incorrect time zone: posixrules"] [stack="github.com/pingcap/parser/terror.MustNil\n\t/home/jenkins/agent/workspace/tidb_v4.0.7/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20200921063432-e220cfcfd026/terror/terror.go:257\nmain.createStoreAndDomain\n\t/home/jenkins/agent/workspace/tidb_v4.0.7/go/src/github.com/pingcap/tidb/tidb-server/main.go:259\nmain.main\n\t/home/jenkins/agent/workspace/tidb_v4.0.7/go/src/github.com/pingcap/tidb/tidb-server/main.go:179\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"] [stack="github.com/pingcap/parser/terror.MustNil\n\t/home/jenkins/agent/workspace/tidb_v4.0.7/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20200921063432-e220cfcfd026/terror/terror.go:257\nmain.createStoreAndDomain\n\t/home/jenkins/agent/workspace/tidb_v4.0.7/go/src/github.com/pingcap/tidb/tidb-server/main.go:259\nmain.main\n\t/home/jenkins/agent/workspace/tidb_v4.0.7/go/src/github.com/pingcap/tidb/tidb-server/main.go:179\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"]`

### 4. What is your TiDB version? (Required)

4.0.7

NA

My 2 cents:

Issue seems to be caused by the fact that certain time-zones have a symlink to posixrules.

TiUP go script seems to be recursively reading /etc/localtime and then when a timezone ends at a symlink to posixrules, TiDB is unable to figure out the actual timezone as 'posixrules' is not a valid time zone.

Seems to be similar to this issue:
HowardHinnant/date#252

Had the same problem when running tiup playground to start tidb component on the same machine.

**Comment by shenoy-anurag** [comment]
I tried to change the timezone to other time-zones to check whether it'll work but it doesn't. 
It throws the same error.

I switched server timezone to my local timezone and tidb ran on my local without issues, but that that also somehow throws the same error.
(btw I rebooted the server and double checked the new timezone using `timedatectl` but still no dice. Also checked which timezone `/etc/localtime` was pointing to.)

So far, I've checked with 4 different timezones.
